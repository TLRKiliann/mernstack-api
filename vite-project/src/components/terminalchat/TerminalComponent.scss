.terminal {
  width: 90%;
  height: 90%;
  margin: auto;
  box-shadow: 0px 0px 4px slategray;
  border: 1px solid slategray;
  border-radius: 15px 15px 0px 0px;

  .div--terminal {
    width: 100%;
    height: calc(100% - 40px);
    margin: auto;
    display: flex;
    flex-direction: column;
    background-color: rgba(10,10,100, 0.9);
    border-radius: 15px 15px 0px 0px;
    overflow: scroll;
    overflow-x: hidden;


    .intro--terminal {
      width: auto;
      
      height: 60px;

      margin-left: 10px;
      margin-right: 10px;
      display: flex;
      align-items: center;
      font-family: "Junction", sans-serif;
      font-weight: bold;
      color: orange;
      border-bottom: 1px dashed slategray;
    }


    .map--msg {
      width: auto;
      height: auto;
      max-width: 100%;
      padding-top: 10px;
      padding-right: 10px;
      padding-left: 10px;
      display: flex;
      flex-direction: column;

      .para--chat {
        margin-top: 5px;
        margin-bottom: 0px;
        text-align: justify;
        text-justify: inter-word;
        font-family: "DejaVu Sans", sans-serif;
        font-size: 0.9rem;
        color: lightgreen;
      }

      .legend--date {
        margin-top: 0px;
        font-size: 0.7rem;
        color: slategray;
      }
    }
  }
  .subterminal {
    width: 100%;
    height: 40px;
    display: flex;
    align-items: center;
    justify-content: space-between;
    background-color: transparent;
    border-top: 1px solid slategray;

    .div--psubterminal {
      position: absolute;
      margin-top: 0px;
      margin-left: 0px;

      .arrow {
        margin-left: 10px;
        color: lightgreen;
      }
    }

    .input--message {
      width: 80%;
      padding: 8px 45px;
      font-size: 1.0rem;
      background-color: rgba(10,10,100, 0.6);
      border: none;
      outline: none;
      box-shadow: 0px 0px 7px whitesmoke;
      color: lightgreen;
    }

    .btn--user {
      padding: 16px 10px;
      margin: auto 20px;
      margin-top: -6px;
      background: rgba(10, 10, 100, 0.4);
      border: none;
      outline: none;
      border-radius: 50%;
      box-shadow: 0px 0px 4px whitesmoke;
      font-weight: bold;
      color: #88ec88;
      z-index: 4;

      &:hover {
        transform: scale(1.1);
        background: rgba(10, 10, 100, 0.2);
        color: orange;
        cursor: pointer;
      }
      &:active {
        transform: scale(1.1);
        color: yellow;
      }
    }
  }
}