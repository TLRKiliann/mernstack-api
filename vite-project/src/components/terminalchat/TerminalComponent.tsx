import React, { useState, useEffect } from 'react'
import { useAuthLogin } from '../../context/AuthProvider'
import { db_users } from '../../models/db_users'
import { UserType } from '../../models/usertype'
import './TerminalComponent.scss'


interface TerminalProps {
  roomStyle: object
}

interface UsernameProps {
  username: string
}

const TerminalComponent: React.FC = (props: {TerminalProps, UsernameProps}) => {

  const [message, setMessage] = useState<string>("")
  const [messages, setMessages] = useState<Array<string>>(JSON.parse(localStorage.getItem("Messages")))

  useEffect(() => {
    localStorage.setItem("Messages", JSON.stringify(messages))
  }, [messages])

  const { username } = useAuthLogin()

  const handleInput = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.preventDefault()
    if (message) {
      setMessages([...messages, {id: Date().slice(0, 24),
        usr: `${username}`, msg: `${message} ✉`}])
    }
    setMessage("")
  }

  return(
    <section className="terminal">

      <div className="div--terminal">

        <span data-testid="spantestid" className="intro--terminal">
          {props.roomStyle}
        </span>
        
        {messages?.map((data) => (
          <div key={data?.id} className="map--msg">
            <p className="para--chat">$ ▶ {data?.usr} ~ {data?.msg}</p>
              <span className="legend--date">{data?.id}</span>
          </div>
          ))
        }
      </div>
      
      <div className="subterminal">
        
        <div className="div--psubterminal">
          <p className="arrow">$ ▶</p>
        </div>

        <input
          type="text"
          value={message}
          onChange={(e) => setMessage(e.target.value)}
          className="input--message"
          placeholder="Write something here..."
          required
        />
        
        <button
          type="button"
          onClick={handleInput}
          className="btn--user"
        >
          Enter
        </button>

      </div>

    </section>
  )
}

export default TerminalComponent