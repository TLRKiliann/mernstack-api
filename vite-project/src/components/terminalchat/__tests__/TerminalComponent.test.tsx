import React from "react";
import { screen, render } from '@testing-library/react';
import { expect, vi } from 'vitest'
import { create } from 'react-test-renderer'
import '@testing-library/jest-dom'
import TerminalComponent from "../TerminalComponent.tsx";

test('TerminalComponent testid', () => {
  render(<TerminalComponent />);
  const mytestid = screen.getByTestId("terminaltest");
  expect(mytestid).toBeInTheDocument();
})

test('MatchSnapShot test TerminalComponent', () => {
  const treeTerminalComponent = create(<TerminalComponent />)
  expect(treeTerminalComponent.toJSON()).toMatchSnapshot()
})