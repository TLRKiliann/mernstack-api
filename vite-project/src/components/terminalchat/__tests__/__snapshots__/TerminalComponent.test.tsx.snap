// Vitest Snapshot v1

exports[`MatchSnapShot test TerminalComponent 1`] = `
<section
  className="terminal"
  data-testid="terminaltest"
>
  <span
    className="intro--terminal"
  >
    <h3
      className="intro--terminalh3"
    />
  </span>
  <div
    className="div--terminal"
  />
  <div
    className="subterminal"
  >
    <div
      className="div--psubterminal"
    >
      <p
        className="arrow"
      >
        â–¶
      </p>
    </div>
    <input
      className="input--message"
      onChange={[Function]}
      placeholder="Write something here..."
      required={true}
      type="text"
      value=""
    />
    <button
      className="btn--user"
      onClick={[Function]}
      type="button"
    >
      Enter
    </button>
  </div>
</section>
`;
